FrostyEditor/Windows/MainWindow.xaml.cs

```diff
@@ -386,19 +386,25 @@ private void launchButton_Click(object sender, RoutedEventArgs e)
             // setup ability to cancel the process
             CancellationTokenSource cancelToken = new CancellationTokenSource();
 
-            Random r = new Random();
-            string editorModName = $"EditorMod{r.Next(1000, 9999).ToString("D4")}.fbmod";
             launchButton.IsEnabled = false;
 
             // get all mods
             List<string> modPaths = new List<string>();
-
+            // Remove existing mods, because unhandled exceptions will cause mods to remain
             DirectoryInfo modDirectory = new DirectoryInfo($"Mods/{ProfilesLibrary.ProfileName}");
             foreach (string modPath in Directory.EnumerateFiles($"Mods/{ProfilesLibrary.ProfileName}/", "*.fbmod", SearchOption.AllDirectories))
-                modPaths.Add(Path.GetFileName(modPath));
+                File.Delete(modPath);
+
+            DateTime localDate = DateTime.Now;
+            Random r = new Random();
+            string editorModName = $"EditorMod_{r.Next(1000, 9999).ToString("D4")}.fbmod";
+            while (File.Exists(editorModName))
+            {
+                editorModName = $"EditorMod_{r.Next(100000, 999999).ToString("D6")}.fbmod";
```

Comment:
why use 6 numbers here instead of 4

Reply:
I'm sorry, I'm afraid it will be repeated (there are more than 10 mod left in my editor), you can change it back to 4.